### PR TITLE
Golang Homework 4.1 sort.Interface

### PR DESCRIPTION
**Pull request by stevenstr**
Implement sort.Interface interface for People type.
 *				Sort people in order of increasing age.
 *				If two people have the same age - sort them by name.
 *				Add test cases.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	return p[i].lastName < p[j].lastName
+}
+
+//Swap method
+func (p People) Swap(i, j int) {
+	p[i], p[j] = p[j], p[i]
+}
+
+//Interface interface which implement some methods from gplang.org/pkg
+type Interface interface {
```
Comment (wshaman) [inline main.go]: It's better to use more meaningful name, like `Sorter`

File: main.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+//Less method
+func (p People) Less(i, j int) bool {
+
+	if p[i].birthDay.Sub(p[j].birthDay) > 0 {
+		return true
+	} else if p[i].birthDay.Sub(p[j].birthDay) < 0 {
+		return false
+	} else if p[i].firstName < p[j].firstName {
```
Comment (wshaman) [inline main.go]: this shouldn't be `else if`, just a new `if`. This func seems to be sophisticated, but still it's OK.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	// Less reports whether the element with
+	// index i should sort before the element with index j.
+	Less(i, j int) bool
+	// Swap swaps the elements with indexes i and j.
+	Swap(i, j int)
+}
+
+func main() {
+	ivanIvanovDate1, err := time.Parse("2006-Jan-02", "2005-Aug-10")
+	if err != nil {
```
Comment (wshaman) [inline main.go:60]: this code can be extracted as a separate func to increase readability, like 
```go
func exitOnErr(err error) {
    if err == nil {
        return
    }
    fmt.Println(err)
    os.Exit(1)
}
```

File: main.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		os.Exit(1)
+	}
+	artiomIvanovDate, err := time.Parse("2006-Jan-02", "2005-Aug-10")
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	//create array of test data
+	p := People{
+		{"Ivan", "Ivanov", ivanIvanovDate1},
```
Comment (wshaman) [inline main.go:76]: setting private parameters like this is not the best solution, sometimes it'll lead to ambiguous behavior
WDYT on adding a creation func near the `Person`? like
```go
func NewPerson(first, last string, bDate time.Time) {
    return  Person {
	firstName: first
	lastName:  last
	birthDay:  bDate
    }
}
```

**Review by wshaman — COMMENTED** [review]
Just a few minor tweaks to go.

File: main.go (hunk 1 above)
Comment (imatsiash-ib) [reply main.go]: This interface is already defined within the `sort` package. Why we need to redefine it here?

Comment (stevenstr) [reply main.go]: > This interface is already defined within the `sort` package. Why we need to redefine it here?
YES. Done!)

File: main.go (hunk 2 above)
Comment (stevenstr) [reply main.go]: Thanks! Done:)

**Review by wshaman — APPROVED** [review]
Nice!

**Review by imatsiash-ib — APPROVED** [review]

**Review by burov — APPROVED** [review]
Looks ok to me

**Pull request closed without merging**